data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -63,15 +56,29 @@ public void run() {
         }
     }
 
+    int processSqsMessages() {
```

Comment:
This is definitely an improvement and as such I'd be happy to take it. But, I also recommend that this indicates a new class would more appropriate. Then you don't have to write unit tests which verify package protected methods.

Reply:
I like the suggestion as well.